### ISSUE TITLE
[Feature Request] GoogleFont Optimization with variable instead of static strings

### ISSUE THREAD

**Issue by dohomi** [vercel/next.js#24639]
### What version of Next.js are you using?

10.2.0

### What version of Node.js are you using?

14.x

### What browser are you using?

Chrome

### What operating system are you using?

macOS

### How are you deploying your application?

next start

### Describe the Bug

Hello
I am using following approach to render a font based on a variable which does not work as expected:
```js
          <link href={getFontBasedOnSetting(settings)} rel="stylesheet" />
```
This code results in two <link /> tags inside of <head>
```html
<link rel="stylesheet" data-href="https://fonts.googleapis.com/css2?family=Roboto:wght@400">
<link rel="stylesheet" href="https://fonts.googleapis.com/css2?family=Roboto:wght@400">
```
So basically the link is attached twice, one with data-href and one with href. Is it not allowed to have a function call to determine the font string?

### Expected Behavior

Inline CSS instead of a <link /> tag

### To Reproduce

Instead of writing the href directly into the link tag have some dynamic string generation.

**Comment by joe-bell** [comment]
The link tags are [statically analysed](https://twitter.com/timneutkens/status/1387428092780613635) – so unfortunately variables aren't supported

**Comment by dohomi** [comment]
oh shoot then I can not make any use of it. Then this bug needs to be transferred to a feature request, because I think that many people will set their fonts either in env vars or some other locations to be more dynamic.

**Comment by dohomi** [comment]
I took now a quite naive approach and fetch the definition in page props via `getFontDefinitionFromNetwork` and inject it with style myself. Is that a reasonable approach?

**Comment by lachlanjc** [comment]
With the new next/font API, can you decide in code which of the font variables to use at runtime & obviate this issue? I would not recommend setting fonts in environment variables for this reason.

**Comment by leerob** [comment]
Indeed, this should be solved now with `@next/font` 😄 

https://nextjs.org/docs/basic-features/font-optimization
